crates/sui-rosetta/src/types.rs

```diff
@@ -0,0 +1,821 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
is it a good idea to include the rosetta types in sui ? https://www.rosetta-api.org/docs/versioning.html#a-note-about-new-sdks per recommendation here maybe a separate repo would be easier to maintain versioning

Reply:
This is good idea, but I will leave this for later.....

I am reusing some of the Sui's types in rosetta's type to reduce codes needed for conversion, currently it's all handcrafted, we might want to consider code gen from schema in the future.